clang/include/clang/AST/Decl.h

```diff
@@ -283,15 +283,16 @@ class NamedDecl : public Decl {
   /// Creating this name is expensive, so it should be called only when
   /// performance doesn't matter.
   void printQualifiedName(raw_ostream &OS) const;
-  void printQualifiedName(raw_ostream &OS, const PrintingPolicy &Policy) const;
+  void printQualifiedName(raw_ostream &OS, const PrintingPolicy &Policy,
+                          bool WithGlobalNsPrefix = false) const;
 
   /// Print only the nested name specifier part of a fully-qualified name,
   /// including the '::' at the end. E.g.
   ///    when `printQualifiedName(D)` prints "A::B::i",
   ///    this function prints "A::B::".
   void printNestedNameSpecifier(raw_ostream &OS) const;
-  void printNestedNameSpecifier(raw_ostream &OS,
-                                const PrintingPolicy &Policy) const;
+  void printNestedNameSpecifier(raw_ostream &OS, const PrintingPolicy &Policy,
+                                bool WithGlobalNsPrefix = false) const;
```

Comment:
Could this be a member of the Printing Policy instead?  It seems to fit better there than in a separate parameter.

Reply:
I could look into that but this is how fully qualified type names are currently handled in clang. That is where we got this idea from. I had spoken to Prem about this. https://clang.llvm.org/doxygen/namespaceclang_1_1TypeName.html#aa956321ff24d3b3d1265b8ee4dc031c0

I'm not sure how to add this to the printing policy. I could look into that as a followup patch if necessary if you're ok with that.

Reply:
Since this is the way it gets done elsewhere, I think this is fine.